golang/channel.go

```diff
@@ -140,10 +142,24 @@ func NewChannel(serviceName string, opts *ChannelOptions) (*Channel, error) {
 		connectionOptions: opts.DefaultConnectionOptions,
 		log:               logger.WithFields(LogField{"service", serviceName}),
 		statsReporter:     statsReporter,
-		traceReporter:     traceReporter,
 		handlers:          &handlerMap{},
 		subChannels:       &subChannelMap{},
 	}
+
+	var traceReporter TraceReporter
+
```

Comment:
nit: avoid too many blank lines, and avoid branching where possible. A more idiomatic way to write this would be:
```go
traceReporter := opts.TraceReporter
if opts.TraceReporterFactory != nil {
  traceReporter = opts.TraceReporterFactory(ch)
}
if traceReporter == nil {
  traceReporter = NullReporter
}
ch.SetTraceReporter(traceReporter)
```